dash-renderer/src/APIController.react.js

```diff
@@ -19,18 +20,29 @@ class UnconnectedContainer extends Component {
     constructor(props) {
         super(props);
         this.initialization = this.initialization.bind(this);
+        this.emitReady = this.emitReady.bind(this);
         this.state = {
             errorLoading: false,
         };
+
+        // Event emitter to communicate when the DOM is ready
+        this.events = new EventEmitter();
+        // Flag to determine if we've really updated the dash components
+        this.renderedTree = false;
```

Comment:
This is probably not an issue here but the APIController instance, like any component instance, could be reused for another APIController and the ctor will not be run a 2nd time. As such it would be safer to assign `this.renderedTree = false` in `componentWillMount` instead.

Reply:
As discussed with @alexcjohnson the above is not an issue. Only components with `key` now get reused.